pkg/telemetry/cassandra_metrics_filters.go

```diff
@@ -0,0 +1,50 @@
+package telemetry
+
+import (
+	"strings"
+
+	telemetry "github.com/k8ssandra/k8ssandra-operator/apis/telemetry/v1alpha1"
+	"github.com/k8ssandra/k8ssandra-operator/pkg/cassandra"
+	v1 "k8s.io/api/core/v1"
+)
+
+var (
+	DefaultFilters = []string{"deny:org.apache.cassandra.metrics.Table",
+		"deny:org.apache.cassandra.metrics.table",
+		"allow:org.apache.cassandra.metrics.table.live_ss_table_count",
+		"allow:org.apache.cassandra.metrics.Table.LiveSSTableCount",
+		"allow:org.apache.cassandra.metrics.table.live_disk_space_used",
+		"allow:org.apache.cassandra.metrics.table.LiveDiskSpaceUsed",
+		"allow:org.apache.cassandra.metrics.Table.Pending",
+		"allow:org.apache.cassandra.metrics.Table.Memtable",
+		"allow:org.apache.cassandra.metrics.Table.Compaction",
+		"allow:org.apache.cassandra.metrics.table.read",
+		"allow:org.apache.cassandra.metrics.table.write",
+		"allow:org.apache.cassandra.metrics.table.range",
+		"allow:org.apache.cassandra.metrics.table.coordinator",
+		"allow:org.apache.cassandra.metrics.table.dropped_mutations"}
+)
+
+// InjectCassandraTelemetryFilters adds MCAC filters to the cassandra container as an env variable.
+// If no filters are defined and telemetry was set in the CRD, the default filters are used.
+func InjectCassandraTelemetryFilters(telemetrySpec *telemetry.TelemetrySpec, dcConfig *cassandra.DatacenterConfig) {
+	if telemetrySpec == nil || telemetrySpec.Prometheus == nil {
+		return
+	}
+
+	metricsFilters := make([]string, len(telemetrySpec.Prometheus.McacMetricFilters))
+	containerIndex, containerFound := cassandra.FindContainer(dcConfig.PodTemplateSpec, "cassandra")
+
+	if len(telemetrySpec.Prometheus.McacMetricFilters) > 0 {
```

Comment:
Should we make it possible to NOT filter metrics by passing an empty `McacMetricFilters` slice? 

`McacMetricFilters` would need to become a slice pointer to accomodate a true nil value.

Reply:
I'd still change it to a nil pointer because that's more idiomatic for the k8s API. Nil means "do the default", empty means "empty filters". This means that the user needs less knowledge of how the filtering/MCAC config file works.

Reply:
idiomatic we go then 👍

Reply:
With all due respect, I am not aware of `*[]string` being idiomatic for Kubernetes APIs. In fact I don't see much of that in Go in general. I do see `[]*Foo` some. With Kubernetes I almost always see `[]string` or `[]Foo` for example. 

If you can point me to an API conventions doc or some references/example that demonstrate otherwise that would be great.

Reply:
@Miles-Garnsey I stand corrected. I found the doc I was looking for along with another good reference:

* https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md
* https://github.com/kubernetes/kubernetes/issues/34641

Ideally, I'd rather avoid the pointers, but I understand that it allows us to distinguish between default vs no filters.